components/browser/awesomebar/src/main/java/mozilla/components/browser/awesomebar/layout/DefaultSuggestionViewHolder.kt

```diff
@@ -58,6 +64,12 @@ internal sealed class DefaultSuggestionViewHolder {
                 suggestion.onSuggestionClicked?.invoke()
                 selectionListener.invoke()
             }
+
+            val hasTitle = !title.isNullOrBlank()
+            editView.visibility = if (hasTitle) View.VISIBLE else View.GONE
+            editView.setOnClickListener {
```

Comment:
Yeah. When I tested this using a local build I found that the append didn't work in Fenix and probably because we need to make changes to the provider's handling in here: https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt

There are some instructions you can [follow here](https://github.com/mozilla-mobile/fenix#auto-publication-workflow-for-android-components-and-application-services) that would help you test your local A-C changes against a local Fenix build as well.

Reply:
So this is a blocker, right?

Reply:
Yes, I'm going to re-test it again soon and I'll get back if I find something. 🙂

Reply:
Ok, so nothing more for me to do, right?

Reply:
If you're interested, you could also try to run a local AC build with Fenix as well to figure out what is causing the bug.

Reply:
Will do.

Reply:
Unfortunately I can't seem to be able to build with autopublish.

I'm getting a bunch of compilation errors, primarily 

```
Supertypes of the following classes cannot be resolved. Please make sure you have the required dependencies in the classpath:
    class mozilla.components.browser.menu.view.MenuButton, unresolved supertypes: mozilla.components.concept.menu.MenuButton
```

Reply:
Did you try rebasing this AC PR so that it matches against Fenix? Fenix master builds off of a nightly version of AC so there can be API mismatches if a PR is stale.

Reply:
Yeah, rebased everything on master :/

Reply:
It appears that `concept-menu` is really missing from the fenix' dependencies.

Reply:
Yeah, adding `concept-menu` to the dependencies did the trick.

Reply:
The issue is `awesomeBar.setOnEditSuggestionListener` is never called. fenix doesn't seem to use `AwesomeBarFeature`.

Reply:
Ok, the fix is trivial:

```
Index: app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt	(revision 6c0be8db1d5292dcf4d3e6cc0f88a755b98cec4d)
+++ app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt	(date 1595629355648)
@@ -169,6 +169,8 @@
             )
         )
 
+        awesomeBarView.view.setOnEditSuggestionListener(toolbarView.view::setSearchTerms)
+
         val urlView = toolbarView.view
             .findViewById<InlineAutocompleteEditText>(R.id.mozac_browser_toolbar_edit_url_view)
         urlView?.importantForAccessibility = View.IMPORTANT_FOR_ACCESSIBILITY_NO
```

![fenix](https://user-images.githubusercontent.com/1191816/88440386-14ad7f00-ce0e-11ea-9f5e-eada483254da.gif)

Reply:
Great! Do you want to open a PR in Fenix to land that change there too?